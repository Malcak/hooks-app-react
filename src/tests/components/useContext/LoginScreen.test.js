import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/useContext/LoginScreen';
import { UserContext } from '../../../components/useContext/UserContext';

describe('tests on the <LoginScreen /> component', () => {
  const setUser = jest.fn();
  const user = { id: '1234', name: 'Fernando', email: 'fernando@mail.com' };
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );
  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should call the setUser function with the expected argument', () => {
    wrapper.find('button').simulate('click');
    expect(setUser).toHaveBeenCalledWith(user);
  });
});
