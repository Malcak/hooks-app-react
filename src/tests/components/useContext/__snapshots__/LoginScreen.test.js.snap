// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <LoginScreen /> component should display the component correctly 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
